capnp/src/primitive_list.rs

```diff
@@ -97,6 +111,15 @@ impl <'a, T> Builder<'a, T> where T: PrimitiveElement {
     pub fn set(&mut self, index: u32, value: T) {
         PrimitiveElement::set(&self.builder, index, value);
     }
+
+    #[cfg(all(target_endian = "little"))]
+    pub fn to_slice(&self) -> &mut [T] {
+        let bytes = self.builder.into_raw_bytes();
+        unsafe {
```

Comment:
`self.builder.element_size != T::element_size()` is possible in this case too. (For example, we could have an `any_pointer::Builder` and initialize it as a struct list but then call `get_as()` as a primitive list.)

Reply:
That's a really weird use, but possible. :D 
I made this perform the same check and return an option as well.